openff/toolkit/topology/molecule.py

```diff
@@ -2107,10 +2113,10 @@ def is_in_ring(self):
                 "This Bond does not belong to a Molecule object"
             )
 
-        for ring in self._molecule.rings:
-            if self.atom1.molecule_atom_index in ring:
-                if self.atom2.molecule_atom_index in ring:
-                    return True
+        if self.atom1.is_in_ring(toolkit_registry=toolkit_registry):
+            if self.atom2.is_in_ring(toolkit_registry=toolkit_registry):
+                return True
+
```

Comment:
This logic does not ensure that each atom is in _the same_ ring, only that it is is _some_ ring. A diphenyl is a simple tests case for this; depending the criteria there are 12 or 13 bonds in rings. Which definition should be used here?

Reply:
Hmmm... I think the "12" definition should be used. So we may need to delegate to these calls:
* https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html?highlight=bond%20ring#rdkit.Chem.rdchem.Bond.IsInRing
* https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemClasses/OEBondBase.html#OEChem::OEBondBase::IsInRing